coderd/agentapi/api.go

```diff
@@ -114,6 +119,24 @@ func New(opts Options) *API {
 		WorkspaceID:              opts.WorkspaceID,
 	}
 
+	// Don't cache details for prebuilds, though the cached fields will eventually be updated
+	// by the refresh routine once the prebuild workspace is claimed.
+	if !workspace.IsPrebuild() {
+		api.cachedWorkspaceFields = CachedWorkspaceFields{
+			ID:             workspace.ID,
+			OwnerID:        workspace.OwnerID,
+			OrganizationID: workspace.OrganizationID,
+			TemplateID:     workspace.TemplateID,
+			Name:           workspace.Name,
+			OwnerUsername:  workspace.OwnerUsername,
+			TemplateName:   workspace.TemplateName,
+			AutostartSchedule: sql.NullString{
+				String: workspace.AutostartSchedule.String,
+				Valid:  workspace.AutostartSchedule.Valid,
+			},
+		}
```

Comment:
Given there is a mutex in the CachedWorkspaceFields, we should probably always assign to a pointer. 

In the `api` struct that is. So we need an `else` condition to init an empty Cache